### ISSUE TITLE
`if` is not allowed in a `const fn`

### ISSUE THREAD

**Issue by AndreKR** [DoumanAsh/error-code#1]
I'm compiling this crate because it's a dependency of the [`clipboard-win`](https://crates.io/crates/clipboard-win) crate that I'm using.

It used to work alright, but today I'm re-compiling and I get an error message:

```
   Compiling error-code v2.0.1
error[E0658]: `if` is not allowed in a `const fn`
   --> c:\dev-tools\Rust\.cargo\registry\src\github.com-1ecc6299db9ec823\error-code-2.0.1\src\lib.rs:141:13
    |
141 | /             if self.code == 10035 {
142 | |                 return true;
143 | |             }
    | |_____________^
    |
    = note: see issue #49146 <https://github.com/rust-lang/rust/issues/49146> for more information
    = help: add `#![feature(const_if_match)]` to the crate attributes to enable
```

I'm on `rustc 1.45.0-nightly (f8d394e51 2020-05-05)`.

**Comment by DoumanAsh** [comment]
ah this is stable in 1.46

Do you not upgrade your nightly often?
I normally only follow 1.46
Although I guess I can revert https://github.com/DoumanAsh/error-code/commit/4d6f3ee984a5a5ceb1c0b7b2a03e1e6a87e568e3

**Comment by AndreKR** [comment]
Well, actually I wanted to simply upgrade my nightly, but I ran into [another issue](https://github.com/rust-lang/rust/issues/76984) with that.

I opened this issue because I figured something like that is probably a breaking change or requires some kind of feature gate or something like that? (I'm quite new to Rust, so I don't know for sure.)

**Comment by AndreKR** [comment]
As far as I'm concerned I found a nightly now (`nightly-2020-08-29`) where both issues are ok.

**Comment by DoumanAsh** [comment]
yeah I'm normally lazy so I don't really worry about backward compatibility.
I can revert code and do not use `if` in `const fn` if you want so let me know

**Comment by DoumanAsh** [comment]
Reverted it for now, released as `2.0.2`
